src/Network/Minio/Utils.hs

```diff
@@ -171,23 +172,33 @@ http req mgr = do
 
   return resp
   where
-    tryHttpEx :: (R.MonadResourceBase m) => m a
+    tryHttpEx :: (MonadUnliftIO m) => m a
               -> m (Either NC.HttpException a)
-    tryHttpEx = ExL.try
+    tryHttpEx = UEx.try
     contentTypeMay resp = lookupHeader Hdr.hContentType $ NC.responseHeaders resp
 
 -- Similar to mapConcurrently but limits the number of threads that
 -- can run using a quantity semaphore.
-limitedMapConcurrently :: (MonadIO m, R.MonadBaseControl IO m)
+limitedMapConcurrently :: MonadUnliftIO m
                        => Int -> (t -> m a) -> [t] -> m [a]
 limitedMapConcurrently count act args = do
```

Comment:
we need to handle the case where `count` could be zero. One way to do it is,
`limitedMapConcurrently 0 _ _ = return []`

Reply:
Fixed.